src/ripple/shamap/SHAMap.h

```diff
@@ -183,10 +183,19 @@ class SHAMap
             std::function<void(std::shared_ptr<SHAMapItem const> const&)> const&) const;
 
     // comparison/sync functions
+
+    /** Check for nodes in the SHAMap not available
+
+        Traverse the SHAMap efficiently, maximizing I/O
+        concurrency, to discover nodes referenced in the
+        SHAMap but not available locally.
+
+        @param maxNodes The maximum number of found nodes to return
+        @param filter The filter to use when retrieving nodes
+        @param return The nodes known to be missing
+    */
     std::vector<std::pair<SHAMapNodeID, uint256>>
-    getMissingNodes (
-        std::size_t max,
-        SHAMapSyncFilter *filter);
+    getMissingNodes (int maxNodes, SHAMapSyncFilter *filter);
```

Comment:
I think earlier versions of getMissingNodes could return a few more nodes than requested. Passing in the limit as a signed value prevented it from underflowing if it was decremented past zero in cases like that.